src/odin/narrativebuilder.cc

```diff
@@ -1199,27 +1358,47 @@ std::string NarrativeBuilder::FormVerbalUturnInstruction(Maneuver& maneuver,
 
   // Determine which phrase to use
   uint8_t phrase_id = 0;
-  if (!street_names.empty()) {
-    phrase_id += 1;
-    if (maneuver.to_stay_on()) {
+  std::string junction_name;
+  std::string guide_sign;
+
+  if (maneuver.HasGuideSign()) {
+    // Set the toward phrase - it takes priority over street names and junction name
+    phrase_id = 7;
+    // Assign guide sign
+    guide_sign = maneuver.signs().GetGuideString(element_max_count, limit_by_consecutive_count, delim,
+                                                 maneuver.verbal_formatter());
+  } else if (maneuver.HasJunctionNameSign()) {
+    // Set the junction phrase - it takes priority over street names
+    phrase_id = 6;
+    // Assign guide sign
+    junction_name =
+        maneuver.signs().GetJunctionNameString(element_max_count, limit_by_consecutive_count, delim,
+                                               maneuver.verbal_formatter());
+  } else {
+    if (!street_names.empty()) {
       phrase_id += 1;
+      if (maneuver.to_stay_on()) {
+        phrase_id += 1;
```

Comment:
Why do we do "add" to `phrase_id` in some places, and set it to a specific value in others? Seems like `phrase_id` is an index into a map, so not sure why use the `+=` here and other places.

Reply:
we add when a certain attribute can map to multiple phrase values depending on other attributes - in this case a non empty street name can map to 1, 2, 4, or 5

Reply:
Ok, thanks for the clarification.